buildSrc/src/main/groovy/org/apache/groovy/gradle/GroovyLibraryExtension.groovy

```diff
@@ -64,16 +66,18 @@ class GroovyLibraryExtension {
     final TaskContainer tasks
     final ProjectLayout layout
 
+    @Inject
     GroovyLibraryExtension(ObjectFactory factory,
+                           ProjectLayout layout,
                            SharedConfiguration sharedConfiguration,
                            JavaPluginConvention javaPluginConvention,
                            JavaPluginExtension javaPluginExtension,
                            SoftwareComponentContainer components,
                            ConfigurationContainer configurations,
-                           TaskContainer tasks,
-                           ProjectLayout layout
+                           TaskContainer tasks
     ) {
         this.objects = factory
+        this.layout = layout
         this.sharedConfiguration = sharedConfiguration
         this.includeInGroovyAll = factory.property(Boolean).convention(true)
         this.grooid = factory.property(Boolean).convention(false)
```

Comment:
I think you should get away with declaring the `grooid` property abstract and then calling the abstract getter in the constructor setting it to a convention. This would allow you to get rid of using the `ObjectFactory`, if you wouldn't need it to instantiate the attributes.

So something like this should work:

```
class GroovyLibraryExtension {
    abstract Property<Boolean> getGrooid()

    GroovyLibraryExtension(...) {
        grooid.convention(false)
    }
}
```

Reply:
I could but given that I need to inject other properties in any case, I prefer keeping the things consistent and not have a mix of abstract properties and injected properties